web/api/queries/query.c

```diff
@@ -1516,7 +1516,7 @@ void rrdr_fill_tier_gap_from_smaller_tiers(RRDDIM *rd, size_t tier, time_t now)
 
             if(sp.end_time > latest_time_t) {
                 latest_time_t = sp.end_time;
-                store_metric_at_tier(rd, tr, t, sp, sp.end_time * USEC_PER_SEC);
+                store_metric_at_tier(rd, tier, t, sp, sp.end_time * USEC_PER_SEC);
```

Comment:
This was the fault. So, I renamed `tr` to `read_tier` to make it clear what it is.